File: src/isitcg.web/Models/Rule.cs
using System.Collections.Generic;

namespace isitcg
{
    public class Rule
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public string Result { get; set; }
        public string BlogUrl { get; set; }
        public IEnumerable<string> Ingredients { get; set; }
    }
}

File: internal/isitcg/rule.go
package isitcg

import (
	"encoding/json"
	"io/ioutil"
)

type rules struct {
	Rules []Rule
}

type Rule struct {
	Name        string
	Description string
	Result      string `yaml:"Result"`
	BlogUrl     string
	Rank        int      `yaml:"Rank"`
	Ingredients []string `yaml:"Ingredients"`
}

func LoadRules(filename string) ([]Rule, error) {
	b, err := ioutil.ReadFile(filename)
	if err != nil {
		return nil, err
	}
	var rules rules
	err = json.Unmarshal(b, &rules)
	if err != nil {
		return nil, err
	}
	return rules.Rules, err
}
